tests/hats_import/verification/generate_malformed_catalogs.py

```diff
@@ -0,0 +1,197 @@
+import random
+import shutil
+from pathlib import Path
+
+import attrs
+import pyarrow
+import pyarrow.dataset
+import pyarrow.parquet
+
+DATA_DIR = Path(__file__).parent.parent.parent.parent / "tests/data"
+INPUT_CATALOG_DIR = DATA_DIR / "small_sky_object_catalog"
+
+
+def run(input_catalog_dir: Path = INPUT_CATALOG_DIR, data_dir: Path = DATA_DIR) -> None:
```

Comment:
In general, test data is generated via this notebook, so that if we make other format changes, we can quickly re-generate all unit test data in one go.

https://github.com/astronomy-commons/hats-import/blob/main/tests/data/generate_data.ipynb

So there are a few options for this PR (in no particular order):

1 - incorporate into the existing notebook
2 - create a second notebook in that same directory so folks can easily find it when re-generating unit test data
3 - skip for this PR, and maybe move it in the future.

Reply:
This kind of thing is generally the trickiest part of writing verification (making data that's bad enough to trip the checks), so I really appreciate that you took the time to do it, and make it reproducible!!

Reply:
I like option 3!